syfertext/span.py

```diff
@@ -0,0 +1,309 @@
+        if vector_count == 0:
+            span_vector = self.doc.vocab.vectors.default_vector
+        else:
+            # Create the final span vector
+            span_vector = vectors / vector_count
+
+        return span_vector
+
+    def get_vector(self, excluded_tokens: Dict[str, Set[object]] = None):
+        """Get Span vector as an average of in-vocabulary token's vectors,
+        excluding token according to the excluded_tokens dictionary.
+
+        Args
+            excluded_tokens (Dict): A dictionary used to ignore tokens of the document based on values
+                of their attributes, the keys are the attributes names and they index, for efficiency, sets of values.
+                Example: {'attribute1_name' : {value1, value2},'attribute2_name': {v1, v2}, ....}
+
+        Returns:
+            span_vector: Span vector ignoring excluded tokens
+        """
+        # If the excluded_token dict in None all token are included
+        if excluded_tokens is None:
+            return self.vector
+
+        # Enforcing that the values of the excluded_tokens dict are sets, not lists.
+        excluded_tokens = {
+            attribute: set(excluded_tokens[attribute]) for attribute in excluded_tokens
+        }
+
+        vectors = None
+
+        # Count the tokens that have vectors
+        vector_count = 0
+
+        for token in self:
+
+            # Get the vector of the token if one exists and if token is not excluded
+
+            include_token = True
+
+            include_token = all(
+                [
+                    getattr(token._, key) not in excluded_tokens[key]
+                    for key in excluded_tokens.keys()
+                    if hasattr(token._, key)
+                ]
+            )
+
+            if token.has_vector and include_token:
+                # Increment the vector counter
+                vector_count += 1
+
+                # Cumulate token's vector by summing them
+                vectors = token.vector if vectors is None else vectors + token.vector
+
+        # If no tokens with vectors were found, just get the default vector(zeros)
+        if vector_count == 0:
+            span_vector = self.doc.vocab.vectors.default_vector
+        else:
+            # Create the final span vector
+            span_vector = vectors / vector_count
+        return span_vector
+
+    def as_doc(self):  # tags: List[str] = None, description: str = None):
+        """Create a `Doc` object with a copy of the `Span`'s tokens.
+
+            Returns (Doc or DocPointer):
+                The new `Doc` copy (or id to `Doc` object) of the span.
+        """
+
+        # Handle circular imports
+        from .doc import Doc
+
+        # Create a new doc object on the required location
+        # Assign the same owner on which this object resides
+        doc = Doc(self.doc.vocab, owner=self.owner)
+
+        # Client of the doc created will be same as the span's client
+        doc.client = self.client
```

Comment:
We can but I just wanted to highlight the importance of this step. So that when someone else reads this, it will be easy for them.

Reply:
Actually, it will be cleaner to put it in the Doc constructor, this will make the documentation of the Doc __init__ function more complete. You can still put a comment before the __init__ call to expalain the idea

Reply:
Done.